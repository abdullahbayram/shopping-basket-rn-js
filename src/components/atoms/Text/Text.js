import * as React from 'react';
import PropTypes from 'prop-types';
import { Text as PaperText } from 'react-native-paper';

const Text = ({ children, variant = 'bodyMedium' }) => <PaperText variant={variant}>{children}</PaperText>;

Text.propTypes = {
  children: PropTypes.node.isRequired,
  variant: PropTypes.string,
};

export default Text;
