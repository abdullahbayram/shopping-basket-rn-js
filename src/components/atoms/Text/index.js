import * as React from 'react';
import PropTypes from 'prop-types';
import { Text as PaperText, useTheme } from 'react-native-paper';

const Text = ({ children, variant = 'bodyMedium', style, numberOfLines, ellipsizeMode, testID = 'text' }) => {
  const theme = useTheme();
  return (
    <PaperText
      testID={testID}
      style={[{ color: theme.colors.tertiary }, style]}
      variant={variant}
      numberOfLines={numberOfLines}
      ellipsizeMode={ellipsizeMode}
    >
      {children}
    </PaperText>
  );
};

Text.propTypes = {
  children: PropTypes.node.isRequired,
  variant: PropTypes.string,
  testID: PropTypes.string,
  ellipsizeMode: PropTypes.string,
  numberOfLines: PropTypes.number,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
};

export default Text;
