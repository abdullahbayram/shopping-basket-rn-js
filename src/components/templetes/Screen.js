import React from 'react';
import { StyleSheet, View } from 'react-native';
import PropTypes from 'prop-types';

const Screen = ({ children }) => {
  return <View style={styles.container}>{children}</View>;
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    paddingLeft: 16,
    paddingRight: 9,
  },
});

export default Screen;

Screen.propTypes = {
  children: PropTypes.node.isRequired,
};
