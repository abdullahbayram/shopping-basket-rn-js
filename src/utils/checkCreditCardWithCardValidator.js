const valid = require('card-validator');

const checkCreditCardWithCardValidator = (creditCardNumber) => {
  const numberValidation = valid.number(creditCardNumber);
  console.log(numberValidation, 'numberValidation');
  return numberValidation.isValid;
};

export default checkCreditCardWithCardValidator;
