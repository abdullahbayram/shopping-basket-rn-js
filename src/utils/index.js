export { default as fetchWithTimeout } from './fetchWithTimeout';
export { default as showToast } from './showToast';
export { default as paymentUtils } from './paymentUtils';
