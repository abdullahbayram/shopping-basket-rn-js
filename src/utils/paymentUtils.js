import { strings } from '../constants';

export default {
  filterNumericInput: (text) => text.replace(/[^0-9]/g, ''),
  formatExpirationDate: (value) => {
    const sanitized = value.replace(/[^0-9]/g, '');
    if (sanitized.length <= 2) return sanitized;
    return `${sanitized.slice(0, 2)}/${sanitized.slice(2, 4)}`;
  },
  getIcon: (field, errors, isCreditCardValid) => {
    console.log(errors, 'errors');
    const { icons } = strings; // Extract icons from strings
    let isValid = !errors[field]; // Check if the field is valid

    if (field === 'creditCardNumber') {
      isValid = isCreditCardValid;
    }
    return isValid ? icons[`${field}Check`] : icons[field]; // Dynamically map field to valid/invalid icons
  },
};
