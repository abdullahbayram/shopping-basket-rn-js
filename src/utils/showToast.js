import { Alert } from 'react-native';

const showToast = ({ title = 'Notice', message = 'Something happened' } = {}) => {
  if (!title && !message) {
    console.log('Invalid input: title and message cannot both be empty');
    return;
  }
  Alert.alert(title, message);
};

export default showToast;
