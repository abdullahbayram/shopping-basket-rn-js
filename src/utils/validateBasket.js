const validateBasket = (array) => !!array && Array.isArray(array) && array.length > 0;

export default validateBasket;
