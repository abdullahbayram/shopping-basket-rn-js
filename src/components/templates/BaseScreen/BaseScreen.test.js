import React from 'react';
import { screen } from '@testing-library/react-native';
import { renderInThemeProvider } from '../../../../__tests__/utils/renderInThemeProvider';
import Screen from '.';
import Text from '../../atoms/Text';

describe('<Screen />', () => {
  it('renders correctly and matches the snapshot', () => {
    const { toJSON } = renderInThemeProvider(
      <Screen>
        <Text>Test Content</Text>
      </Screen>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    renderInThemeProvider(
      <Screen>
        <Text>Test Content</Text>
      </Screen>,
    );

    expect(screen.getByText('Test Content')).toBeTruthy();
  });

  it('applies styles from the theme correctly', () => {
    const theme = {
      colors: { background: 'blue' },
    };

    renderInThemeProvider(
      <Screen testID="base-screen">
        <Text>Styled Content</Text>
      </Screen>,
      theme,
    );

    const baseScreen = screen.getByTestId('base-screen');

    expect(baseScreen.props.style).toEqual(
      expect.objectContaining({
        flex: 1,
        backgroundColor: 'blue',
      }),
    );
  });
});
