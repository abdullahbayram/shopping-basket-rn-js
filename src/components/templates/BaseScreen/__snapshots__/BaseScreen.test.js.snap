// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screen /> renders correctly and matches the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFF",
      "flex": 1,
      "paddingLeft": 16,
      "paddingRight": 9,
      "paddingTop": 10,
    }
  }
  testID="base-screen"
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0.25,
            "lineHeight": 20,
          },
          [
            {
              "color": "#34300F",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Test Content
  </Text>
</View>
`;
